### PR TITLE
Decorate

### PR DESCRIPTION
**Pull request by roshan-bajgain**
## OOP school library: decorate a class
- Create a class Nameable.
    - Implement a method called correct_name that will raise a NotImplementedError.
- Turn your Person class to Nameable
- Prepare base Decorator
- Prepare CapitalizeDecorator and TrimmerDecorator

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+require './corrector'
+
 class Person
```
Comment (MahmoudBakr23) [inline person.rb]: - [x] Please keep in mind that the `Person` class has to inherit from the `Namabale` class, and in this case, I don't see the `Namabale` class in your files, unless it is the `Corrector` class!

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @roshan-bajgain 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Create the `Corrector` which is the `Namabale` class
However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️

- [x] Please create `Decorator`, `CapitalizeDecorator`,  and `TrimmerDecorator` classes

_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (roshan-bajgain) [reply person.rb]: Thank you for your suggestion. I have implemented all the changes told by you.

Comment (rloterh) [reply person.rb]: This issue is well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Roshan, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:. Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**